File: Estructurales Java/Adapter.java
// Adapter Pattern
class Target {
    public String request() {
        return "Target: The default target's behavior.";
    }
}

class Adaptee {
    public String specificRequest() {
        return ".eetpadA eht fo roivaheb laicepS";
    }
}

class Adapter extends Target {
    private Adaptee adaptee;
    
    public Adapter(Adaptee adaptee) {
        this.adaptee = adaptee;
    }
    
    @Override
    public String request() {
        String result = adaptee.specificRequest();
        return "Adapter: (TRANSLATED) " + new StringBuilder(result).reverse().toString();
    }
}

// Usage
class AdapterDemo {
    public static void main(String[] args) {
        Target target = new Target();
        System.out.println(target.request());
        
        Adaptee adaptee = new Adaptee();
        Adapter adapter = new Adapter(adaptee);
        System.out.println(adapter.request());
    }
}

File: Estructurales Python/adapter.py
# Adapter Pattern
class Target:
    def request(self):
        return "Target: The default target's behavior."

class Adaptee:
    def specific_request(self):
        return ".eetpadA eht fo roivaheb laicepS"

class Adapter(Target):
    def __init__(self, adaptee: Adaptee):
        self.adaptee = adaptee
    
    def request(self):
        return f"Adapter: (TRANSLATED) {self.adaptee.specific_request()[::-1]}"

# Usage
if __name__ == "__main__":
    target = Target()
    print(target.request())
    
    adaptee = Adaptee()
    adapter = Adapter(adaptee)
    print(adapter.request())
